warden/x/async/keeper/plugin.go

```diff
@@ -0,0 +1,85 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	types "github.com/warden-protocol/wardenprotocol/warden/x/async/types/v1beta1"
+)
+
+// deductKeychainFees is to deduct fees from creator's address to warden escrow address.
+func (k Keeper) deductPluginFees(
+	ctx context.Context,
+	pluginUser sdk.AccAddress,
+	fees types.PluginFee,
+) (*types.DeductedFee, error) {
```

Comment:
afaict we usually return types.DeductedFee (not a pointer) in other functions so I would keep it that way

```suggestion
) (types.DeductedFee, error) {
```